src/Manager.php

```diff
@@ -1,87 +1,64 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Keboola\OAuthV2Api;
 
-/**
- *
- */
+use InvalidArgumentException;
+
 class Manager extends Common
 {
-    protected $requiredApiDetails = [
-        "component_id",
-        "friendly_name",
-        "app_key",
-        "app_secret",
-        "auth_url",
-        "token_url",
-        "oauth_version"
+    private const REQUIRED_API_DETAILS = [
+        'component_id',
+        'friendly_name',
+        'app_key',
+        'app_secret',
+        'auth_url',
+        'token_url',
+        'oauth_version',
     ];
 
-    public function __construct($manageToken, $config = [])
+    public function __construct(string $manageToken, array $config)
     {
-        $this->client = $this->getClient(['X-KBC-ManageApiToken' => $manageToken], $config);
+        parent::__construct(['X-KBC-ManageApiToken' => $manageToken], $config);
     }
 
-    /**
-     * @param array $details ## more than that!
-     * @return array|object
-     */
-    public function add(array $details)
+    public function add(array $details): array
     {
         $this->validateApiDetails($details);
-
-        return $this->apiPost("manage", [
-            'form_params' => $details
```

Comment:
Toto je dost podivný - to API je json, takže si to vysvětluju jedině tak, že to nikdy nefungovalo